### PR TITLE
[Monitoring] Add new amp agent

### PR DESCRIPTION
**Pull request by freignat91**
Add working new amp-agent, part of new monitoring stack
- manage docker events to maintain on memory containers list
- for each container:
   - start a log stream and push logs in nats (amp-logs subject)
   - store the last log timestamp in order to do not resend all logs in case of crash
   - start a metrics stream and push metrics in nats (amp-metrics subject)

### CODE REVIEW

**Comment by subfuzion** [comment]
Closing and reopening in attempt to force Travis CI to perform build checks.

**Review by subfuzion — COMMENTED** [review]
@freignat91 Thanks for getting this updated. I also pitched in on some refactoring, formatting, and linting. Please review to make sure you approve of changes.

**Comment by freignat91** [comment]
Updates look good to me. I tested them, it's ok

File: agent/agent.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+			nb = 0
+			a.nbLogs = 0
+			a.nbMetrics = 0
+		}
+		time.Sleep(time.Duration(conf.period) * time.Second)
+	}
+}
+
+// Starts logs and metrics stream of eech new started container
+func (a *Agent) uddateStreams() {
```
Comment (ndegory) [inline agent/agent.go]: should it be called updateStreams?

File: agent/config.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+func (cfg *AgentConfig) loadConfigUsingEnvVariable() {
+	cfg.dockerEngine = getStringParameter("DOCKER", cfg.dockerEngine)
+	cfg.natsURL = getStringParameter("NATS_URL", cfg.natsURL)
+	cfg.apiPort = getStringParameter("API_PORT", cfg.apiPort)
+	cfg.period = getIntParameter("PERIOD", cfg.period)
+	cfg.clientID = getStringParameter("CLIENTID", cfg.clientID)
+	cfg.clusterID = getStringParameter("CLIENTID", cfg.clusterID)
+}
+
+// Display amp-pilot configuration
```
Comment (ndegory) [inline agent/config.go]: incorrect comment

File: agent/config.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	log.Printf("Docker-engine: %s\n", conf.dockerEngine)
+	log.Printf("Nats URL: %s\n", conf.natsURL)
+	log.Printf("ClientId: %s\n", conf.clientID)
+	log.Printf("ClusterId: %s\n", conf.clusterID)
+	log.Println("----------------------------------------------------------------------------")
+}
+
+// Return env variable value, if empty return default value
+func getStringParameter(envVariableName string, def string) string {
+	value := os.Getenv(envVariableName)
```
Comment (ndegory) [inline agent/config.go]: LookupEnv would be better than Getenv, to make the difference between unset and empty values.

**Review by ndegory — APPROVED** [review]
few comments, but LGTM

**Pull request merged**